server/src/test/java/com/objectcomputing/checkins/services/checkinnotes/CheckinNoteControllerTest.java

```diff
@@ -0,0 +1,272 @@
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.MutableHttpRequest;
+import io.micronaut.http.client.HttpClient;
+import io.micronaut.http.client.annotation.Client;
+import io.micronaut.http.client.exceptions.HttpClientResponseException;
+import io.micronaut.test.annotation.MicronautTest;
+import io.micronaut.test.annotation.MockBean;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+
+@MicronautTest
+public class CheckinNoteControllerTest {
+
+    @Inject
+    @Client("/services/checkin-note")
+    HttpClient client ;
+
+    @Inject
+    private CheckinNoteServices checkinNoteServices;
+
+    @MockBean(CheckinNoteServices.class)
+    public CheckinNoteServices checkinNoteServices() {
+        return mock(CheckinNoteServices.class);
+    }
+
+    @Test
+    void testCreateCheckinNote(){
+     CheckinNoteCreateDTO checkinNoteCreateDTO = new CheckinNoteCreateDTO();
+     checkinNoteCreateDTO.setCheckinid(UUID.randomUUID());
+     checkinNoteCreateDTO.setCreatedbyid(UUID.randomUUID());
+     checkinNoteCreateDTO.setDescription("test");
+
+     CheckinNote cNote = new CheckinNote(checkinNoteCreateDTO.getCheckinid(), checkinNoteCreateDTO.getCreatedbyid(), checkinNoteCreateDTO.getDescription());
+
+     when(checkinNoteServices.save(eq(cNote))).thenReturn(cNote);
+
+     final HttpRequest<CheckinNoteCreateDTO> request = HttpRequest.POST("", checkinNoteCreateDTO);
+     final HttpResponse<CheckinNote> response = client.toBlocking().exchange(request, CheckinNote.class);
+
+     assertEquals(cNote, response.body());
+     assertEquals(HttpStatus.CREATED, response.getStatus());
+     assertEquals(String.format("%s/%s", request.getPath(), cNote.getId()), response.getHeaders().get("location"));
+
+    }
+
+    @Test
+    void testCreateInvalidCheckinNote(){
+     CheckinNoteCreateDTO checkinNoteCreateDTO = new CheckinNoteCreateDTO();
+
+     CheckinNote cNote = new CheckinNote(UUID.randomUUID(),UUID.randomUUID(), "test");
+
+     when(checkinNoteServices.save(eq(cNote))).thenReturn(cNote);
+
+     final HttpRequest<CheckinNoteCreateDTO> request = HttpRequest.POST("", checkinNoteCreateDTO);
+     HttpClientResponseException responseException = assertThrows(HttpClientResponseException.class,
+     () -> client.toBlocking().exchange(request, Map.class));
+
+     JsonNode body = responseException.getResponse().getBody(JsonNode.class).orElse(null);
+     JsonNode errors = Objects.requireNonNull(body).get("_embedded").get("errors");
+     JsonNode href = Objects.requireNonNull(body).get("_links").get("self").get("href");
+     List<String> errorList = List.of(errors.get(0).get("message").asText(), errors.get(1).get("message").asText())
+             .stream().sorted().collect(Collectors.toList());
+     assertEquals("checkinnote.checkinid: must not be null", errorList.get(0));
```

Comment:
this should be `checkinNote.checkinid: must not be null`
Same with next line capitalize the N in note.

This test case will fail as is.